snippets/csharp/VS_Snippets_Misc/tpl_cancellation/cs/snippet02.cs

```diff
@@ -27,19 +27,19 @@ static void Main()
                 }
 
             }
-        }, tokenSource2.Token); // Pass same token to StartNew.
+        }, tokenSource2.Token); // Pass same token to Task.Run.
 
         tokenSource2.Cancel();
 
-        // Just continue on this thread, or Wait/WaitAll with try-catch:
+        // Just continue on this thread, or await with try-catch:
         try
         {
-            task.Wait();
+            await task;
         }
         catch (AggregateException e)
```

Comment:
By changing to `await`, the `AggregateException` isn't the right exception type. It should be an `OperationCanceledException`. 

That also means the `foreach` loop isn't needed, and a single `Console.WriteLine` is sufficient.

Reply:
Good point! I caught that in the other example. Now it's fixed for this one too!